### ISSUE TITLE
Different payload format via direct dispatch vs via form

### ISSUE THREAD

**Issue by alexey-pelykh** [afitiskin/redux-saga-routines#40]
https://github.com/afitiskin/redux-saga-routines/blob/1cf1594b4dd24602bbd40b2586ea3af9ff906625/src/bindRoutineToReduxForm.js#L6

This adds some extra complexity that 'callee' input format differs depending on a 'caller'. What are your thoughts on that?

**Comment by alexey-pelykh** [comment]
This issue has bigger impact in Typescript, where
```
export const action = createRoutine<IPayload, IPayload>('CREATE_DEMO_ENTITY', _.identity)
```
gives a hint about format of the payload, yet writing
```
function* handler ({ payload }: Action<IPayload>) {
```
seems impossible or at least incorrect, since there's even no definition of the `values` + `props` payload

**Comment by afitiskin** [comment]
Do not do magic, just dispatch `action.trigger` with same payload format as `bindRoutineToReduxForm` do :)

**Comment by afitiskin** [comment]
There is no other way to pass all data that `redux-form` gives to us (they are all useful in some cases).

**Comment by alexey-pelykh** [comment]
Also true :| messy mess :)

**Comment by afitiskin** [comment]
For me it looks strange when you manually dispatch `trigger` for routine that bound to `redux-form`...  It looks like some hack, so, when you do "hack" you need to follow the rules (in the case – use special scheme for your action's payload)

**Comment by alexey-pelykh** [comment]
Hm, that could be, but not my particular the case: a simple form to add an email to a list, payload needs `listId` ("hidden" field via `initialValues`), and `email` field. That thing is sent via `addEmail` routine that is callable not only via form, in case of "add from contact" story.

**Comment by afitiskin** [comment]
I prefer to separate form and data handling. So, in my opinion, it’s better to have 2 sagas and 2 routines: first for form handling (validation, etc) and second for data handling: send data to API, etc.
First saga triggers second one in case if validation passed. Adding from contacts directly triggers second one via specific action.

**Comment by afitiskin** [comment]
Handling two different scenarios using one routine and saga is messy mess :)

**Comment by alexey-pelykh** [comment]
I would argue a little bit about that :), especially in case when the only difference in saga is `(payload.values || payload)`. Though I got your point! Anyhow, to avoid confusion in `bindRoutineToReduxForm` I'd adjust a definition over here https://github.com/afitiskin/redux-saga-routines/blob/master/src/index.d.ts#L91 to define `Routine<T>` as something more specific with well-defined payload according to actual thing over here https://github.com/afitiskin/redux-saga-routines/blob/master/src/bindRoutineToReduxForm.js#L6

What are your thoughts on that?

**Comment by alexey-pelykh** [comment]
(if you're open to that, I'd add that into PR #41 )

**Comment by afitiskin** [comment]
Do it, I will merge

**Comment by alexey-pelykh** [comment]
I've updated the #41 so that now Payload has to explicitly have `values` & `props`

**Comment by alexey-pelykh** [comment]
Closing as #41 was merged and published as `redux-saga-routines@2.2.4`
